Add kubectl-gs usage data collection info (#2310)
giantswarm/docs: src/content/vintage/use-the-api/kubectl-gs/telemetry.md

last_review_date: 2024-10-16
---

Since kubectl-gs version 4.3.0, Giant Swarm collects anonymous usage data about `kubectl gs` command execution. This allows us to understand better which commands are used how widely and how frequently, and to better prioritize our development efforts.

You are free to [opt out](#optout) of this data collection. However, we appreciate your support and ensure that we only collect non-identifyable information and keep data collection to a minimum.

## Details we collect

Whenever a command is executed (this does not include calling a command with `-h` or `--help`), we collect the following details:

- Name of the command executed
- kubectl-gs version
- Name of the operating system
- Processor architecture (e.g. amd64, arm64)
- The [library](https://github.com/giantswarm/telemetrydeck-go) and version used to submit the data (e. g. `telemetrydeck-go/0.0.1`)
- A generated user identifier hash (more details [below](#user-id-hash))
- A random UUID as a session identifier. Currently this is unique per command execution. We may change this in the future to correlate several data submissions to a single session.

## User identifier hash {#user-id-hash}

> We submit a hash that is supposed to be unique to each user using kubectl-gs, to allow counting unique users. The hash is generated based on the following data and SHA256-hashed before submission:

Reviewer: ```suggestion
We submit a SHA256 hash along with the telemetry sent from kubectl-gs, to allow counting unique users. This hash is generated based on the following data:
```
